Microsoft.Toolkit.Uwp.UI.Controls.DataGrid/DataGrid/DataGrid.cs

```diff
@@ -3199,7 +3199,7 @@ public void ScrollIntoView(object item, DataGridColumn column)
             }
             else
             {
-                int slot = -1;
+                int slot;
```

Comment:
Worried about changing the default value from -1 here as this is about a row index. Think this should stay -1 as it's checked later in the passed ScrollSlotIntoView function [here](https://github.com/windows-toolkit/WindowsCommunityToolkit/blob/master/Microsoft.Toolkit.Uwp.UI.Controls.DataGrid/DataGrid/DataGridRows.cs#L652)

Reply:
`Slot` will always have a value set to `slot = rowGroupInfo.Slot;` (if branch) or `slot = SlotFromRowIndex(rowIndex);` (else branch) hence the IDE warning. The force initialization here is not needed since the value will be overridden just after.
I can add a `#pragma` to locally disable the warning if we want to be very protective but usage of uninitialized variable is well reported by VS so it doesn't feel to be needed.

---
In reply to: [390619125](https://github.com/windows-toolkit/WindowsCommunityToolkit/pull/3155#discussion_r390619125) [](ancestors = 390619125)